### PR TITLE
Converted saphyr and pacbio samples and libraries to use vuex and rem…

### PR DESCRIPTION
**Pull request by stevieing**
…oved the use for mattype.

### CODE REVIEW

File: src/views/saphyr/SaphyrSamples.vue [hunk 1]
```diff
@@ -155,7 +152,8 @@ export default {
     },
     async provider() {
       try {
-        this.items = await this.getMaterial(consts.MAT_TYPE_REQUESTS)
+        await this.setRequests()
+        this.items = this.requests
```
Comment (harrietc52) [inline src/views/saphyr/SaphyrSamples.vue]: We could possibly get rid of items in this component and call this.requests from the b-table, but looks like the clearPreFilter() method is using items. Although not sure how that works, if it does!?

File: src/views/saphyr/SaphyrLibraries.vue [hunk 2]
```diff
@@ -152,7 +151,7 @@ export default {
     // Get all the libraries
     // Provider function used by the bootstrap-vue table component
     async provider() {
-      this.items = await this.getMaterial(consts.MAT_TYPE_LIBRARIES)
+      this.items = await this.setLibraries()
```
Comment (harrietc52) [inline src/views/saphyr/SaphyrLibraries.vue]: Maybe a more vuex'y way is for the provider to use call setLibraries(), then have a separate getter the get the libraries. PacbioRuns.vue does a similar thing

File: src/views/pacbio/PacbioLibraries.vue [hunk 3]
```diff
@@ -143,7 +142,7 @@ export default {
     // Get all the libraries
     // Provider function used by the bootstrap-vue table component
     async provider() {
-      this.items = await this.getMaterial(consts.MAT_TYPE_LIBRARIES)
+      this.items = await this.setLibraries()
```
Comment (harrietc52) [inline src/views/pacbio/PacbioLibraries.vue]: Same here as below comment

File: src/store/traction/saphyr/tubes/actions.js [hunk 4]
```diff
@@ -71,19 +71,37 @@ const deleteLibraries = async ({ getters }, libraryIds) => {
...
+  let response = await handlePromise(promise)
+  let libraries = null
+
+  if (response.successful && !response.empty) {
+    libraries = response.deserialize.libraries
+    
+    commit('setLibraries', libraries)
+  }
+
+  return libraries
```
Comment (harrietc52) [inline src/store/traction/saphyr/tubes/actions.js]: may not need to return anything from this function

File: src/store/traction/saphyr/requests/getters.js [hunk 5]
```diff
@@ -0,0 +1,6 @@
+const getters = {
+    requests: state => state.requests,
+    requestsRequest: (state, getters, rootState) => rootState.api.traction.pacbio.requests,
```
Comment (harrietc52) [inline src/store/traction/saphyr/requests/getters.js]: could add a small test for requestsRequest

File: src/store/traction/saphyr/tubes/mutations.js [hunk 6]
```diff
@@ -1,6 +1,9 @@
 const mutations = {
   setTubes (state, tubes) {
     state.tractionTubes = tubes
+  },
```
Comment (harrietc52) [inline src/store/traction/saphyr/tubes/mutations.js]: could add small tests for these, don't think any exist at the moment

File: src/views/saphyr/SaphyrSamples.vue (hunk 1 above)
Comment (stevieing) [reply src/views/saphyr/SaphyrSamples.vue]: Removed pre filter and modified items.

File: src/views/saphyr/SaphyrLibraries.vue (hunk 2 above)
Comment (stevieing) [reply src/views/saphyr/SaphyrLibraries.vue]: Done. Also removed prefilter. Removed unneccessary getters.

File: src/views/pacbio/PacbioLibraries.vue (hunk 3 above)
Comment (stevieing) [reply src/views/pacbio/PacbioLibraries.vue]: Modified to use getter for libraries. Removed clear pre filter. Removed unneeded getters.

File: src/store/traction/saphyr/tubes/actions.js (hunk 4 above)
Comment (stevieing) [reply src/store/traction/saphyr/tubes/actions.js]: I am using the results from the action in the test. Don't know if there is a better way?

File: src/store/traction/saphyr/requests/getters.js (hunk 5 above)
Comment (stevieing) [reply src/store/traction/saphyr/requests/getters.js]: I have added a new issue to review how vuex is setup. I think there are a few getters and other things which are not currently tested.

File: src/store/traction/saphyr/tubes/mutations.js (hunk 6 above)
Comment (stevieing) [reply src/store/traction/saphyr/tubes/mutations.js]: Again I think we are missing tests for a few mutations.

**Pull request merged**